src/Lecture2.hs

```diff
@@ -70,7 +74,15 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [b] -> (Maybe b, [b])
+removeAt i [] = (Nothing, [])
+removeAt i (x:xs)
+    | i < 0 = (Nothing, x:xs)
+    | i == 0 = (Just x, xs)
+    | otherwise = let
+        (maybeZ, zs) = removeAt (i - 1) xs
+        in
+        (maybeZ, x:zs)
```

Comment:
Perfect 👍🏻 

The only potential improvement is to check for negative index only in the beginning and move the rest into a helper recursive function so you won't check on negative indexes on each recursive call